jtransc-rt-core/src/com/jtransc/thread/JTranscThreading.java

```diff
@@ -27,7 +28,10 @@ public void start(Thread thread) {
 			System.err.println("WARNING: Threads not supported! Executing thread code in the parent's thread!");
 			thread.run();
 		}
-
+		
+		@HaxeMethodBody("cpp.vm.Thread.create(function():Void{p0.run__V();});")
+		native public void start_target(Runnable target);
```

Comment:
For the other option suggested below:

You would remove this method. And add an annotation to `start` method:

```
@HaxeMethodBody(target = "cpp", "cpp.vm.Thread.create(function():Void{p0.%{% FIELD java.lang.Thread:target %}.{% METHOD java.lang.Runnable:run %});});")
public void start(Thread thread) {
  if (parent != null) {
    parent.start(thread);
    return;
  }
  System.err.println("WARNING: Threads not supported! Executing thread code in the parent's thread!");
  thread.run();
}
```

In jtransc you can reference private fields like this from outside. When generating code, everything is public to allow this kind of stuff or allowing reflective usage.